starknet_py/net/udc_deployer/deployer.py

```diff
@@ -64,20 +64,52 @@ async def prepare_contract_deployment(
         :param calldata: Constructor args of the contract to be deployed
         :param max_fee: Max amount of Wei to be paid when executing transaction
         :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
-        :return: InvokeFunction
+        :return: NamedTuple with invoke_tx and address fields
```

Comment:
@war-in @cptartur what if we didn't expose `prepare_contract_deployment` here at all and just asked users to provide the call they get here to the execute or sign_invoke_transaction? I think prepare_contract_deployment is pretty useless and duplicates functionality from account. 
It just feels like `create_deployment_call` is the right abstraction here.

Reply:
We could change create_deployment_call to accept an optional `Account` and would use its address.

Reply:
The benefit is that it emphasised that it is just a normal call inside invoke, it becomes obvious how you could atomically do some things after deploying the contract.

Reply:
I don't think it's practical to do so: we're generally using transactions in other parts of the library. As far as i understand, creating calls was only added to the udc for the users that want to create and interact with the account in the same tx - i'd say not the most standard flow.

Reply:
But see what prepare_contract_deployment does now: takes output of `create_deployment_call` and calls `sign_invoke_transaction`. It doesn't do much.

In case user wants to execute the deploy the flow will be the same as now:
```
deploy_call = udc.create_deployment_call(...) # now it is sign_invoke_transaction
account.execute(deploy_call)
```

In case user doesn't want to execute it right away there is just one more function call to do.
```
deploy_call = udc.create_deployment_call(...)
tx = account.sign_invoke_transaction(deploy_call)
```

I really like better composability and emphasis of treating UDC deploy just like any regular call. By hiding this call in tx users won't realise the possibilities. If it is not ergonomic for users then we can add it in the future without any issue.
Also having `auto_estimate` in UDC as a parameter feels like leaking logic from account.

Reply:
Okay, we can do what you suggest and merge.

However I still think we need simplified flow for less advanced users, as the current one is quite a lot more complicated that just `Client.deploy` as it was previously.

Let's create an issue for that.